pkgs/by-name/gr/grayjay/package.nix

```diff
@@ -0,0 +1,117 @@
+{
+  buildDotnetModule,
+  fetchgit,
+  dotnetCorePackages,
+  pkgs,
+  buildNpmPackage,
+  lib,
+}:
+let
+  version = "3";
+  src = fetchgit {
+    url = "https://gitlab.futo.org/videostreaming/Grayjay.Desktop";
+    rev = "refs/tags/${version}";
+    hash = "sha256-gI1M/rDTSH16eeknHJfLA2iawoA5iGTKHfwUW4/lvv8=";
+    fetchSubmodules = true;
+    fetchLFS = true;
+  };
+  frontend = buildNpmPackage {
+    name = "grayjay-frontend";
+    inherit version;
+
+    src = "${src}/Grayjay.Desktop.Web";
+
+    npmBuildScript = "build";
+    npmDepsHash = "sha256-pTEbMSAJwTY6ZRriPWfBFnRHSYufSsD0d+hWGz35xFM=";
+
+    installPhase = ''
+      cp -r dist/ $out
+    '';
+  };
+in
+buildDotnetModule {
+  pname = "grayjay";
+  version = "3";
+
+  inherit src;
+
+  projectFile = [
+    "Grayjay.ClientServer/Grayjay.ClientServer.csproj"
+    "Grayjay.Engine/Grayjay.Engine/Grayjay.Engine.csproj"
+    "Grayjay.Desktop.CEF/Grayjay.Desktop.CEF.csproj"
+    "FUTO.MDNS/FUTO.MDNS/FUTO.MDNS.csproj"
+    "JustCef/DotCef.csproj"
+  ];
+
+  testProjectFile = [
+    "Grayjay.Desktop.Tests/Grayjay.Desktop.Tests.csproj"
+    "Grayjay.Engine/Grayjay.Engine.Tests/Grayjay.Engine.Tests.csproj"
+  ];
+
+  nugetDeps = ./deps.json;
+
+  dotnet-runtime = dotnetCorePackages.aspnetcore_8_0;
+
+  executables = [ "Grayjay" ];
+
+  preBuild = ''
+    rm -r Grayjay.ClientServer/wwwroot/web
+    cp -r ${frontend} Grayjay.ClientServer/wwwroot/web'';
+
+  postInstall = ''
+    chmod +x $out/lib/grayjay/cef/dotcefnative
+    rm $out/lib/grayjay/Portable
+    ln -s /tmp/grayjay-launch $out/lib/grayjay/launch
+    ln -s /tmp/grayjay-cef-launch $out/lib/grayjay/cef/launch
+  '';
+
+  makeWrapperArgs = [
+    "--chdir"
+    "${placeholder "out"}/lib/grayjay"
+  ];
+
+  runtimeDeps = with pkgs; [
```

Comment:
Each of these should be individually brought in instead of using `with pkgs` so that the inputs can be overridden.